File: nodes/core/io/system_prompt_loader_node.py
from typing import Any

from core.types_registry import NodeCategory
from nodes.base.base_node import Base


class SystemPromptLoader(Base):
    """Loads a system prompt from UI-provided content (e.g., uploaded .md/.txt).

    Outputs:
    - system: str
    """

    inputs = {}
    outputs = {"system": str}
    default_params = {"content": ""}
    params_meta = [
        {"name": "content", "type": "textarea", "default": ""},
    ]

    # Keep visible in IO category
    CATEGORY = NodeCategory.IO

    async def _execute_impl(self, inputs: dict[str, Any]) -> dict[str, Any]:
        content = self.params.get("content") or ""
        if not isinstance(content, str):
            content = str(content)
        return {"system": content}

File: backend/src/nodes/core/io/system-prompt-loader-node.ts
// src/nodes/core/io/system-prompt-loader-node.ts
// Translated from: nodes/core/io/system_prompt_loader_node.py

import { Base } from '../../base/base-node';
import { NodeCategory } from '../../../core/types';
import type { NodeInputs, NodeOutputs, ParamMeta, DefaultParams } from '../../../core/types';

/**
 * Loads a system prompt from UI-provided content (e.g., uploaded .md/.txt).
 *
 * Outputs:
 * - system: str
 */
export class SystemPromptLoader extends Base {
  static override inputs: Record<string, unknown> = {};
  static override outputs: Record<string, unknown> = { system: String };
  static override defaultParams: DefaultParams = { content: '' };
  static override paramsMeta: ParamMeta[] = [
    { name: 'content', type: 'textarea', default: '' },
  ];

  // Keep visible in IO category
  static override CATEGORY = NodeCategory.IO;

  protected override async executeImpl(_inputs: NodeInputs): Promise<NodeOutputs> {
    let content = this.params.content ?? '';
    if (typeof content !== 'string') {
      content = String(content);
    }
    return { system: content };
  }
}
